core/templates/dev/head/editor/RouterServices.js

```diff
@@ -110,14 +126,29 @@ oppia.factory('routerService', [
       }
     };
 
-    var _getCurrentStateFromLocationPath = function() {
-      if ($location.path().indexOf('/gui/') !== -1) {
-        return $location.path().substring('/gui/'.length);
+    var _getCurrentStateFromLocationPath = function(pathType) {
+      if (pathType != 'preview' && pathType != 'gui') {
+        return null;
+      }
+      if ($location.path().indexOf('/' + pathType + '/') !== -1) {
+        return $location.path().substring('/' + pathType + '/'.length);
       } else {
         return null;
       }
     };
 
+    var _actuallyNavigate = function(newStateName, pathType) {
```

Comment:
Let's switch the args so we have pathType before newStateName -- that matches the order in which they appear in the URL.